File: algorithms/implementation/designer-pdf-viewer/designer-pdf-viewer-javascript/js/test_designerPdfViewer.js
import {designerPdfViewer} from '../js/designerPdfViewer.js';
import assert from 'assert';

describe('designerPdfViewer', function() {

    it('O tamanho da área destacada é 28', function() {

        let k = [1, 3, 1, 3, 1, 4, 1, 3, 2, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 7];
        let palavra = ['z','a', 'b', 'a'];
        let esperado = 28;

        let resultado = designerPdfViewer(k, palavra)

        assert.strictEqual(resultado, esperado)
    });
});

describe('designerPdfViewer', function() {

    it('O tamanho da área destacada é 9', function() {

        let k = [1, 3, 1, 3, 1, 4, 1, 3, 2, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5];
        let palavra = ['a', 'b', 'c'];
        let esperado = 9;

        let resultado = designerPdfViewer(k, palavra)

        assert.strictEqual(resultado, esperado)
    });
});

describe('designerPdfViewer', function() {

    it('O tamanho da área destacada é 70', function() {

        let k = [6, 5, 7, 3, 6, 7, 3, 4, 4, 2, 3, 7, 1, 3, 7, 4, 6, 1, 2, 4, 3, 3, 1, 1, 3, 5];
        let palavra = ['z', 'b', 'k', 'k', 'f', 'h', 'w', 'p', 'l', 'j'];
        let esperado = 70;

        let resultado = designerPdfViewer(k, palavra)

        assert.strictEqual(resultado, esperado)
    });
});

File: algorithms/implementation/designer-pdf-viewer/designer-pdf-viewer-python/tests/test_designerPdfViewer.py
from designerPdfViewer import designerPdfViewer

class TesteDesignerPdfViewer:
    
    def testeCaso1(self):
        
        h = [1, 3, 1, 3, 1, 4, 1, 3, 2, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 7]
        palavra = 'zaba'
        
        esperado = 28
        
        resultado = designerPdfViewer(h, palavra)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        h = [1, 3, 1, 3, 1, 4, 1, 3, 2, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5]
        palavra = 'abc'
        
        esperado = 9
        
        resultado = designerPdfViewer(h, palavra)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        h = [6, 5, 7, 3, 6, 7, 3, 4, 4, 2, 3, 7, 1, 3, 7, 4, 6, 1, 2, 4, 3, 3, 1, 1, 3, 5]
        palavra = 'zbkkfhwplj'
        
        esperado = 70
        
        resultado = designerPdfViewer(h, palavra)
        
        assert resultado == esperado